src/test/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSinkCmekKeyTest.java

```diff
@@ -0,0 +1,155 @@
+
+    serviceAccountKey = new String(Files.readAllBytes(Paths.get(new File(serviceAccountFilePath).getAbsolutePath())),
+                                   StandardCharsets.UTF_8);
+
+  }
+
+  private GCSBatchSink.GCSBatchSinkConfig getConfig(@Nullable String project, @Nullable String path,
+                                                    @Nullable String serviceAccountType,
+                                                    @Nullable String serviceFilePath,
+                                                    @Nullable String serviceAccountJson) throws NoSuchFieldException {
+    GCSBatchSink.GCSBatchSinkConfig gcsBatchSinkConfig = new GCSBatchSink.GCSBatchSinkConfig();
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("project"), project);
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("path"), path);
+    FieldSetter.setField(gcsBatchSinkConfig, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("format")
+      , "csv");
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCPReferenceSinkConfig.class.getDeclaredField("referenceName")
+        , "testref");
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("serviceAccountType"), serviceAccountType);
+    if (!Strings.isNullOrEmpty(serviceFilePath)) {
+      FieldSetter
+        .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("serviceFilePath"), serviceFilePath);
+    }
+    if (!Strings.isNullOrEmpty(serviceAccountJson)) {
+      FieldSetter
+        .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("serviceAccountJson"), serviceAccountJson);
+    }
+    return gcsBatchSinkConfig;
+  }
+
+  @Test
+  public void testServiceAccountPath() throws Exception {
+    GCSBatchSink.GCSBatchSinkConfig config = getConfig(project, bucketPath, "filePath",
+                                                       serviceAccountFilePath, null);
+    testValidCmekKey(config);
+    testInvalidCmekKey(config);
+  }
+
+  @Test
+  public void testServiceAccountJson() throws Exception {
+    GCSBatchSink.GCSBatchSinkConfig config = getConfig(project, bucketPath, "JSON",
+                                                       null, serviceAccountKey);
+    testValidCmekKey(config);
+    testInvalidCmekKey(config);
+  }
+
+  public void testValidCmekKey(GCSBatchSink.GCSBatchSinkConfig config) throws Exception {
+    MockFailureCollector collector = new MockFailureCollector("gcssink");
+    config.validate(collector);
+    FieldSetter
+      .setField(config, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("location"), "us-east1");
+    FieldSetter
+      .setField(config, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("cmekKey"),
+                "projects/cdf-test-322418/locations/us-east1/keyRings/my_ring/cryptoKeys/test_key");
```

Comment:
is this an actual project or just some value you made up? Don't put references to actual resources in the code.

Reply:
Its just some value I made up to test the cmek key name format.